In [1]:
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import matplotlib.ticker as ticker
import seaborn as sns

import plotly.express as px

%matplotlib inline

In [2]:
data = {'NL' : [2261, 1416, 1041, 681, 427, 271], 'India' : 
        [359, 397, 425, 464, 464, 422], 'Malaysia' : [101, 45, 22, 9, 16, 2], 
        'Sweden': [412, 332, 266, 171, 150, 102]}
# example code data = {'Name':['Tom', 'nick', 'krish', 'jack'], 'Age':[20, 21, 19, 18]} 

new_covid_cases = pd.DataFrame(data)
new_covid_cases

NL  India  Malaysia  Sweden
0  2261    359       101     412
1  1416    397        45     332
2  1041    425        22     266
3   681    464         9     171
4   427    464        16     150
5   271    422         2     102

In [3]:
new_covid_cases['Week'] = [6, 5, 4, 3, 2, 1]
new_covid_cases

NL  India  Malaysia  Sweden  Week
0  2261    359       101     412     6
1  1416    397        45     332     5
2  1041    425        22     266     4
3   681    464         9     171     3
4   427    464        16     150     2
5   271    422         2     102     1

In [4]:
new_covid_cases = new_covid_cases.set_index('Week')
new_covid_cases

NL  India  Malaysia  Sweden
Week                               
6     2261    359       101     412
5     1416    397        45     332
4     1041    425        22     266
3      681    464         9     171
2      427    464        16     150
1      271    422         2     102

In [5]:
new_covid_cases['High_density'] = (new_covid_cases['NL'] + 
new_covid_cases['India']) / 2
new_covid_cases

NL  India  Malaysia  Sweden  High_density
Week                                             
6     2261    359       101     412        1310.0
5     1416    397        45     332         906.5
4     1041    425        22     266         733.0
3      681    464         9     171         572.5
2      427    464        16     150         445.5
1      271    422         2     102         346.5

In [6]:
new_covid_cases['Low_density'] = (new_covid_cases['Malaysia'] + 
new_covid_cases['Sweden']) / 2
new_covid_cases

NL  India  Malaysia  Sweden  High_density  Low_density
Week                                                          
6     2261    359       101     412        1310.0        256.5
5     1416    397        45     332         906.5        188.5
4     1041    425        22     266         733.0        144.0
3      681    464         9     171         572.5         90.0
2      427    464        16     150         445.5         83.0
1      271    422         2     102         346.5         52.0

In [8]:
covid_cases_high_density = new_covid_cases['High_density']
covid_cases_high_density_mean = covid_cases_high_density.mean()
covid_cases_high_density_mean

719.0

#### 1. The Alternative Hypothesis

*   Ha: the average covid cases per million for countries with low population density < 719.  

#### 2. The Null Hypothesis

*  H0: the average covid cases per million  = 719

#### 3. Set Critical Value

*   $\alpha$ = 0.01.

#### 4. Compute the t-score 
We are dealing with an unknown $\sigma$, we need to use Student's t-distribution and hence a t-test. 

In [9]:
# Terms
covid_cases_low_density = new_covid_cases['Low_density']
covid_cases_low_density_mean = covid_cases_low_density.mean()
sample_stdev = covid_cases_low_density.std()
standard_error = sample_stdev / np.sqrt(len(covid_cases_low_density))

In [10]:
t_value = (covid_cases_low_density_mean - covid_cases_high_density_mean) / standard_error
print(t_value)

-18.65903875220958


#### 5. Calculate the p-value

In [11]:
from scipy.stats import t

df = len(covid_cases_low_density) - 1

alpha = 0.01
alpha = 1 - alpha
alpha = t.ppf(alpha, df)
print(alpha)

3.3649299989072747


Since our value for t (18.5) is much higher than the critical value of 3.36, we can reject the null hypothesis. In other words, the average covid cases for countries with a low population density is lower than the average cases with a high population density.